src/dcp/dcp.c

```diff
@@ -429,7 +412,7 @@ int main(int argc, char** argv)
         numpaths_src = numpaths - 1;
     }
 
-    if (numpaths_src == 0 && is_posix_copy) {
+    if (numpaths_src == 0 && !is_daos_copy) {
```

Comment:
@daltonbohning if it is a DAOS copy, and we are using DFS, could numpaths_src here be zero? Would that be an error?

Reply:
@dsikich Oh yes, seg faults. Now that the obj-level copy uses the "paths" as well, we can move that numpaths check back to BEFORE daos_setup() and simplify this a bit.

Reply:
@dsikich I actually found a few things that could be simplified, in my latest commit